jmxtrans-output/jmxtrans-output-logback/src/main/java/com/googlecode/jmxtrans/model/output/LogbackWriter.java

```diff
@@ -98,6 +98,7 @@ public void internalWrite(Server server, final Query query, ImmutableList<Result
 				MDC.put("valuePath", Joiner.on('.').join(result.getValuePath()));
 				MDC.put("Epoch", String.valueOf(result.getEpoch()));
 				log.info("");
+				MDC.clear();
```

Comment:
Clearing the whole MDC is probably not the right thing to do. What if it already has some state, pushed by another piece of code. `MDC.put()` returns a `Closeable` that can be used to clear the specific values.